controllers/ingestendpoint/secret.go

```diff
@@ -111,8 +115,53 @@ func (g *EndpointSecretGenerator) PrepareFields(ctx context.Context, dk *dynatra
 		dataIngestToken = string(token)
 	}
 
+	diUrl, err := getDataIngestUrlFromDk(dk)
+	if err != nil {
+		return nil, err
+	}
+
 	return map[string]string{
-		UrlSecretField:   fmt.Sprintf("%s/v2/metrics/ingest", dk.Spec.APIURL),
+		UrlSecretField:   diUrl,
 		TokenSecretField: dataIngestToken,
 	}, nil
 }
+
+func getDataIngestUrlFromDk(dk *dynatracev1beta1.DynaKube) (string, error) {
```

Comment:
small nitpick:
rename to `getDataIngestUrlFromDynakube` so its clearer

Reply:
renamed